src/vec-raw.md

```diff
@@ -116,15 +118,20 @@ impl<T> Drop for IntoIter<T> {
 impl<T> Vec<T> {
     pub fn into_iter(self) -> IntoIter<T> {
         unsafe {
-            // need to use ptr::read to unsafely move the buf out since it's
+            // need to use `ptr::read` to unsafely move the buf out since it's
             // not Copy, and Vec implements Drop (so we can't destructure it).
             let buf = ptr::read(&self.buf);
             let len = self.len;
             mem::forget(self);
 
             IntoIter {
-                start: *buf.ptr,
-                end: buf.ptr.offset(len as isize),
+                start: buf.ptr.as_ptr(),
+                end: if buf.cap == 0 {
+                    // can't offset off this pointer, it's not allocated!
+                    buf.ptr.as_ptr()
+                } else {
+                    buf.ptr.as_ptr().offset(len as isize)
+                },
```

Comment:
I'm pretty sure this is unnecessary, but it's fair this isn't documented properly.

https://github.com/rust-lang/rust/issues/65108